lib/thesis/api_controller.ex

```diff
@@ -18,6 +19,19 @@ defmodule Thesis.ApiController do
     json conn, %{}
   end
 
+  def import_file(conn, %{"image_url" => ""}), do: json conn, %{path: ""}
+  def import_file(conn, %{"image_url" => image_url, "image_name" => image_name}) do
+    image = HTTPoison.get!(image_url)
+    file = %Plug.Upload{
+      path: %{data: image.body},
+      filename: "imported-" <> Utilities.parameterize(image_url),
+      content_type: (image.headers |> Enum.into(%{}) |> Map.new(fn {k, v} -> {String.downcase(k), v} end))["content-type"]
+    }
+
+    upload_file(conn, %{"file" => file})
+  end
+  def import_file(conn, _), do: json conn, %{path: ""}
+
```

Comment:
adds a new controller action to handle imports. this file constructs a "fake" upload struct from the get response and reuses the existing `upload_file` function. this will be compatible with any uploader